lib/autocomplete-manager.coffee

```diff
@@ -110,26 +113,59 @@ class AutocompleteManager
     return if @isCurrentFileBlackListed()
     cursor = @editor.getLastCursor()
     return unless cursor?
-    cursorPosition = cursor.getBufferPosition()
-    currentScope = cursor.getScopeDescriptor()
-    return unless currentScope?
-    currentScopeChain = currentScope.getScopeChain()
-    return unless currentScopeChain?
-
-    options =
-      editor: @editor
-      buffer: @buffer
-      cursor: cursor
```

Comment:
I'm not suggesting the cursor has a function added to it. I'm suggesting we could additionally provide a getPrefix function on the options object which takes a cursor, and an options object. This is almost better illustrated via a commit that we can revert if needed. I'll have time later to do this.
  
```